memory/jemalloc_nodump_allocator.h

```diff
@@ -85,7 +85,7 @@ class JemallocNodumpAllocator : public BaseMemoryAllocator {
   std::unique_ptr<extent_hooks_t> arena_hooks_;
 
   // Hold thread-local tcache index.
-  ThreadLocalPtr tcache_;
+  mutable ThreadLocalPtr tcache_;
```

Comment:
I think this is fine since `ThreadLocalPtr` is thread-safe, so is the function `GetThreadSpecificCache`. But double check with @igorcanadi